apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// An immutable class representing partition options.
+/// The specified values of `PartitionSizeBytes`, `MaxPartitions` and `CancellationToken`
+/// are utilized in the RPC call while generating the partition tokens.
+/// The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when executing the
+/// partition object. If `DataBoostEnabled` is set to true, the request will be
+/// executed using Spanner independent compute resources.
+/// </summary>
+public sealed class PartitionOptions
+{
+    /// <summary>
+    /// Returns a default instance of <see cref="PartitionOptions"/> with databoost enabled flag set to false.
+    /// </summary>
+    public static PartitionOptions Default { get; } = new PartitionOptions(null, null, false);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the databoost enabled flag updated to the provided value.
```

Comment:
I'd use `<see cref="DataBoostEnabled" />` instead of "the databoost enabled flag".

I'd also avoid saying "update to the provided value" as it could cause confusion about whether it changes this instance. How about:

```csharp
/// Returns a new instance with the same options as this one, but with the specified new value for <see cref="DataBoostEnabled" />.
```

(And ditto the ones below.)